### PR TITLE
envsetup implementation and project rename

### LINKED ISSUE #15: use envsetup instead of global variables
https://pharmaverse.github.io/envsetup/main/index.html

It would be good to promote opensource collaboration between pharma companies. I really like this package which attaches the paths to an environment. We can also use this system to automatically source functions (eg R/functions versus R/programs). I believe we do not need to stick to the strict package structure as long as we can package up everything nicely.

### PR DESCRIPTION
**Pull request by SHAESEN2**
closes #15.

Please check. I had some issues with renv.

### CODE REVIEW

**Comment by robertdevine** [comment]
Best to confirm test coverage with R infrastructure using rspm and latest windows since 2020 Ubuntu 16.x.

**Comment by SHAESEN2** [comment]
@thomas-neitmann @laxamanaj can we please review and merge this branch?

File: renv.lock [hunk 1]
```diff
@@ -2031,6 +2031,29 @@
...
+      "Version": "0.0.1",
+      "Source": "GitHub",
+      "RemoteType": "github",
+      "RemoteHost": "api.github.com",
+      "RemoteUsername": "pharmaverse",
+      "RemoteRepo": "envsetup",
+      "RemoteRef": "main",
+      "RemoteSha": "539514d89cbff4f61635052ba865d05a1d3fb181",
+      "Hash": "c194b17837fb23766fa208f96adfccbc"
+      "Requirements": []
```
Comment (thomas-neitmann) [inline renv.lock:2046]: There's a `,` missing after the hash.

File: DESCRIPTION [hunk 2]
```diff
@@ -23,6 +23,10 @@ Imports:
     tidyr
 Suggests:
     cowplot,
+    visR,
+    envsetup,
+    config
+Suggests:
```
Comment (thomas-neitmann) [inline DESCRIPTION:29]: `Suggests:` is listed twic making the file malformed.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: renv.lock (hunk 1 above)
Comment (SHAESEN2) [reply renv.lock]: done

**Comment by thomas-neitmann** [comment]
@SHAESEN2 The workflows still seem to fail due to an issue with the `renv.lock` file. Can you successfully run `renv::restore()` locally?

**Comment by thomas-neitmann** [comment]
So all workflows fail because at the time `.Rprofile` is executed for the first time all the packages haven't been installed, yet.

**Comment by thomas-neitmann** [comment]
Just pushed a *potential* fix. Let's see if that works.

File: DESCRIPTION [hunk 3]
```diff
@@ -23,6 +23,9 @@ Imports:
     tidyr
 Suggests:
     cowplot,
+    visR,
+    envsetup,
+    config
```
Comment (thomas-neitmann) [inline DESCRIPTION:28]: There's a `,` missing here leading the `R CMD check` workflow to fail.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: .Rprofile [hunk 4]
```diff
@@ -1,7 +1,8 @@
 if (Sys.getenv("GITHUB_ACTIONS") == "") {
   source("renv/activate.R")
-  envsetup::rprofile(config::get(file = file.path(getwd(), "_envsetup.yml")))
+  envsetup::rprofile(config::get(file = file.path(getwd(), "_envsetup.yml"), config = "prod"))
   source("inst/startup.R")
 } else {
   options(repos = "https://cran.microsoft.com/snapshot/2022-11-01")
+  envsetup::rprofile(config::get(file = file.path(getwd(), "_envsetup.yml"), config = "prod"))
```
Comment (thomas-neitmann) [inline .Rprofile:7]: Please remove the new line in the `else` statement as this will make the workflows fail again.

File: DESCRIPTION (hunk 3 above)
Comment (SHAESEN2) [reply DESCRIPTION]: resolved

File: _envsetup.yml [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
-    qc: '/adam/'
-    output: '/output/'
-    programs: '/submission/programs/'
-    functions: '/R/'
+    sdtm: !expr file.path(getwd(), "sdtm")
+    adam: !expr list(PROD = file.path(getwd(), "submission/datasets"), QC = file.path(getwd(), "adam"))
+    output: !expr file.path(getwd(), "output")
+    programs: !expr file.path(getwd(), "submission/programs")
+    functions: !expr file.path(getwd(), "R")
+    envsetup_environ: !expr Sys.setenv(ENVSETUP_ENVIRON = 'QC'); 'QC'
```
Comment (thomas-neitmann) [inline _envsetup.yml:8]: There's no advantage of using `file.path(getwd(), "somepath")` versus just `./somepath` or `somepath` in an RStudio project.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Comment by SHAESEN2** [comment]
How to use this - Imagine adsl is finalized and stored in submission/datasets and adae is not yet created but you want to use it to create adtte. You could rely on the available adae in adam folder.
By using read_path the path is dynamically changed on the availability of the final datasets.

```r
> envsetup::read_path(adam, "adae.xpt")
Read Path:/cloud/project/adam/adae.xpt

[1] "/cloud/project/adam/adae.xpt"
> envsetup::read_path(adam, "adsl.xpt")
Read Path:/cloud/project/submission/datasets/adsl.xpt

[1] "/cloud/project/submission/datasets/adsl.xpt"
```

**Comment by thomas-neitmann** [comment]
Fair enough but how does this relate to the use of `getwd()` when specifying the paths? Does {envsteup} change the working directory?

File: _envsetup.yml (hunk 5 above)
Comment (SHAESEN2) [reply _envsetup.yml]: If that is the preference ... Updated.

File: .Rprofile (hunk 4 above)
Comment (SHAESEN2) [reply .Rprofile]: updated

**Comment by SHAESEN2** [comment]
> Fair enough but how does this relate to the use of `getwd()` when specifying the paths? Does {envsteup} change the working directory?

There is no relationship.

```r 
> adae <- haven::read_xpt(envsetup::read_path(adam, "adae.xpt"))
Read Path:./adam/adae.xpt

> adsl <- haven::read_xpt(envsetup::read_path(adam, "adsl.xpt"))
Read Path:./submission/datasets/adsl.xpt
```

**Comment by SHAESEN2** [comment]
Thanks @thomas-neitmann for the help setting this up :)

**Pull request merged**